homeassistant/components/juicenet/sensor.py

```diff
@@ -17,104 +23,115 @@
 from .const import DOMAIN, JUICENET_API, JUICENET_COORDINATOR
 from .entity import JuiceNetDevice
 
-SENSOR_TYPES = {
-    "status": ["Charging Status", None, None, None],
-    "temperature": [
-        "Temperature",
-        TEMP_CELSIUS,
-        DEVICE_CLASS_TEMPERATURE,
-        STATE_CLASS_MEASUREMENT,
-    ],
-    "voltage": ["Voltage", ELECTRIC_POTENTIAL_VOLT, DEVICE_CLASS_VOLTAGE, None],
-    "amps": [
-        "Amps",
-        ELECTRIC_CURRENT_AMPERE,
-        DEVICE_CLASS_CURRENT,
-        STATE_CLASS_MEASUREMENT,
-    ],
-    "watts": ["Watts", POWER_WATT, DEVICE_CLASS_POWER, STATE_CLASS_MEASUREMENT],
-    "charge_time": ["Charge time", TIME_SECONDS, None, None],
-    "energy_added": ["Energy added", ENERGY_WATT_HOUR, DEVICE_CLASS_ENERGY, None],
-}
+SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key="status",
+        name="Charging Status",
+    ),
+    SensorEntityDescription(
+        key="temperature",
+        name="Temperature",
+        unit_of_measurement=TEMP_CELSIUS,
+        icon="mdi:thermometer",
```

Comment:
Side note: We normally don't need to set a custom icon when setting a device class. The frontend will already use a specific icon per device class.